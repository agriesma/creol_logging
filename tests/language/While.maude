load creol-interpreter .
mod PROGRAM is
  protecting CREOL-SIMULATOR .
  op state : -> State [ctor] .
  eq state = {
    < "Test" : Class | Inh: noInh, Param: noVid, Att: noSubst,
      Mtds: < "init" : Method | Param: noVid, Att: "_" |-> null, Code: skip ;
              return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "l" |-> null, "f" |-> null,
              "_" |-> null, Code: while "/=" ( "l" :: list(emp) ) do
              assign( "f", "l" ; "head" ( "l" ) :: "tail" ( "l" ) ) od  ;
              return ( emp ) >,
      Ocnt: 0 > } .
endm

