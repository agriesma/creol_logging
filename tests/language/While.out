==========================================
reduce in PROGRAM : state .
result State: {
< "Test" : Class | Inh: noInh, Param: noVid, Att: noSubst, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: skip ; return(emp)
    >, < "run" : Method | Param: noVid, Att: "_" |-> null, "f" |-> null, "l"
    |-> null, Code: while "/="("l" :: list(emp)) do assign("f", "l" ; "head"(
    "l") :: "tail"("l")) od ; return(emp) >,
  Ocnt: 0 >
}
Bye.
